Added and updated websites

starring94 opened the pull request:
* [3hentai] added website, resolves #4709
* [anzmanga] added website, resolves #4677
* [ayatoon] added website, resolves #4701
* [bananamecanica] Changed label and domain, resolves #4733
* [baozimh] change page getter function, resolves #4691
* [batoto] added url setter back
* [beetoon] added website, resolves #4623
* [comick] updated website, resolves #4742
* [copymanga] added website
-- resolves #2735
-- resolves #3948
* [dalsei] added website, resolves #4717
* [doujin69] added website, resolves #3921
* [doujinmitai] added website, resolves #4710
* [eclypse] changed place from root to the connectors folder
* [epikmanga] added page getter overwrite function
* [flamescans] Removed connector and image
* [ghostfansub] added website, resolves #4716
* [ichijin-plus] added website, resolves #4721
* [imperfectcomic] changed template from Madara to Mangastream
-- resolves #4455
-- resolves #4664
* [jiangzaitoon] changed domain, resolves #4715
* [komikcast] changed domain, resolves #4729
* [kraw] added website, resolves #4689
* [lelmanga] placed in the correct position, resolves #4685
* [leomanga] changed template from MRCMS to Madara
* [lshiver] added website, resolves #4702
* [manga-italia] added website, resolves #4659
* [manga18-club] fixed page getter
-- resolves #4227
-- resolves #4484
-- resolves #4646
* [mangabox] added website, resolves #87
* [mangacdn] added website
-- resolves #3581
-- resolves #3843
* [mangadna] added website, resolves #4675
* [mangahubru] changed chapter and page getter function
* [mangakawaii] fixed problems, resolves #4692
* [mangaonlineteam] added website, resolves #4629
* [mangaschan] added website, resolves #4736
* [MangaReaderCMS] added clipboard functionality
* [manhwaid] updated domain, resolves #4599
* [manytoon] changed label
* [nhentai] fixed page getter function
* [on-manga] added clipboard functionality
* [opiatoon] changed domain, resolves #4713
* [piedpiperfb] changed domain and image
-- resolves #4714
-- resolves #4738
* [pmscans] changed url and visible name, resolves #4661
* [porncomixonline] changed label
* [readmng] updated website
-- resolves #4643
-- resolves #4663
* [smashscan] added website, resolves #3922
* [tappytoon-de] added website
* [tappytoon-en] added website, resolves #1824
* [tappytoon-fr] added website
-- resolves #4062
-- resolves #4129
* [team1x1] updated website
-- resolves #4576
-- resolves #4671
* [test-app] [ConfigurationDarwin] replaced todo with function in skip
* [test-app] [ConfigurationLinux] replaced todo with function in skip
* [test-app] [ConfigurationWindows] replaced todo with function in skip
* [test-e2e] updated timeout to 50 seconds
* [test-e2e] [comicbrise] removed slash from the end of the chapter ID
* [test-e2e] [kissmanga] removed test
* [test-e2e] [mangadex] updated parameters to v5
* [test-e2e] [readm] updated page matcher
* [test-e2e] [to-corona-ex] changed manga title
* [test-e2e] [to-corona-ex] Changed connector class name
* [test-e2e] [to-corona-ex] Changed accessor from shift to pop
* [to-corona-ex] updated connector
-- standardized for use by Ichijin Plus connector
-- changed to multi-page manga and chapter list
-- reversed chapter order
-- async descrambling
* [tortugaceviri] added website, resolves #4700
* [tredrage] added website, resolves #4703
* [truyentranhaudio] changed template from FM to MojoPortal
* [truyentranhaudioonline] changed template from Madara to TruyenTranhAudio
* [truyenqq] added website, resolves #3883
* [tumanhwas] added website, resolves #4739
* [viewcomics] added website, resolves #3741
* [vinmanga] added website, resolves #4687
* [whitecloudpavilion] changed template from MRCMS to Mangastream
* [yaoifanclube] changed domain, resolves #4734
* [zeurelscan] added website, resolves #4672

starring94 committed:
Added and updated connectors
* added: BeeToon
* added: MangaOnlineTeam
* fixed: ReadMNG
* fixed: ManhwaID

starring94 committed:
Merge branch 'manga-download:master' into s-connectors3

starring94 committed:
Added and updated connectors
* [anzmanga] added website, resolves #4677
* [ayatoon] added website, resolves #4701
* [batoto] added url setter back
* [eclypse] changed place from root to the connectors folder
* [imperfectcomic] changed template from Madara to Mangastream
-- resolves #4455
-- resolves #4664
* [kraw] added website, resolves #4689
* [lelmanga] placed in the correct position, resolves #4685
* [leomanga] changed template from MRCMS to Madara
* [lshiver] added website, resolves #4702
* [manga-italia] added website, resolves #4659
* [manga18-club] fixed page getter
-- resolves #4227
-- resolves #4484
-- resolves #4646
* [mangadna] added website, resolves #4675
* [mangakawaii] fixed problems, resolves #4692
* [MangaReaderCMS] added clipboard functionality
* [on-manga] added clipboard functionality
* [pmscans] changed url and visible name, resolves #4661
* [team1x1] updated website
-- resolves #4576
-- resolves #4671
* [tortugaceviri] added website, resolves #4700
* [tredrage] added website, resolves #4703
* [viewcomics] added website, resolves #3741
* [vinmanga] added website, resolves #4687
* [whitecloudpavilion] changed template from MRCMS to Mangastream
* [zeurelscan] added website, resolves #4672

starring94 committed:
Updated connectors and tests
* [baozimh] changed page getter function, resolves #4691
* [epikmanga] added page getter overwrite function
* [test-app] [ConfigurationDarwin] replaced todo with function in skip
* [test-app] [ConfigurationLinux] replaced todo with function in skip
* [test-app] [ConfigurationWindows] replaced todo with function in skip
* [test-app] [UpdateServerManager] removed link from error
* [test-e2e] updated timeout to 50 seconds
* [test-e2e] [comicbrise] removed slash from the end of the chapter ID
* [test-e2e] [kissmanga] removed test
* [test-e2e] [mangadex] updated parameters to v5
* [test-e2e] [readm] updated page matcher

starring94 committed:
* [test-app] [UpdateServerManager] changed back

starring94 committed:
* [batoto] fixed page getter function

starring94 committed:
Updated connectors
* [flamescans] Removed connector and image
* [mangahubru] changed chapter and page getter function
* [manytoon] changed label
* [porncomixonline] changed label
* [truyentranhaudio] changed template from FM to MojoPortal
* [truyentranhaudioonline] changed template from Madara to TruyenTranhAudio

starring94 committed:
Added and updated connectors
* [3hentai] added website, resolves #4709
* [bananamecanica] Changed label and domain, resolves #4733
* [comick] updated website, resolves #4742
* [copymanga] added website
-- resolves #2735
-- resolves #3948
* [dalsei] added website, resolves #4717
* [doujin69] added website, resolves #3921
* [doujinmitai] added website, resolves #4710
* [ghostfansub] added website, resolves #4716
* [ichijin-plus] added website, resolves #4721
* [jiangzaitoon] changed domain, resolves #4715
* [komikcast] changed domain, resolves #4729
* [mangabox] added website, resolves #87
* [mangacdn] added website
-- resolves #3581
-- resolves #3843
* [mangaschan] added website, resolves #4736
* [nhentai] fixed page getter function
* [opiatoon] changed domain, resolves #4713
* [piedpiperfb] changed domain and image
-- resolves #4714
-- resolves #4738
* [smashscan] added website, resolves #3922
* [test-e2e] changed to-corona-ex manga title
* [to-corona-ex] standardized for use by Ichijin Plus connector
* [truyenqq] added website, resolves #3883
* [tumanhwas] added website, resolves #4739
* [yaoifanclube] changed domain, resolves #4734

starring94 committed:
Added and updated connectors and test
* [baozimh] changed page getter function once again
* [tappytoon-de] added website
* [tappytoon-en] added website, resolves #1824
* [tappytoon-fr] added website
-- resolves #4062
-- resolves #4129
* [test-e2e] [to-corona-ex] Changed connector class name

starring94 committed:
Updated connector
* [comick] Changed page getter function

27136773 commented on src/web/mjs/connectors/Batoto.mjs:
```diff
@@ -32,11 +32,18 @@ export default class Batoto extends AnyACG {
     }
 
+    set url(value) {
+        if (this.config && value) {
```
Leave implementation blank, otherwise the user defined value will be unintentionally overwritten by the application

starring94 replied:
This was copied from other implementations of this

27136773 commented on src/web/mjs/connectors/Batoto.mjs:
```diff
@@ -32,11 +32,18 @@ export default class Batoto extends AnyACG {
     async _getPages(chapter) {
         let script = `
         new Promise(resolve => {
-            const base = JSON.parse(CryptoJS.AES.decrypt(server, batojs).toString(CryptoJS.enc.Utf8));
```
May use `app.items.map(item => item.src || item.isrc);` which contains the decrypted links (after a timeout of ~2500ms)

starring94 replied:
fixed

27136773 commented on src/web/mjs/connectors/ComicK.mjs:
```diff
@@ -20,75 +21,87 @@ export default class ComicK extends Connector {
+            let title = '';
             if(item.vol) {
-                title.push('Vol.', item.vol);
+                title += `Volume ${item.vol} `;
```
Downloaded chapters are identified by their name, when a new naming pattern is applied, user who have downloaded chapters will no longer see them as downloaded...

starring94 replied:
fixed

27136773 commented on src/web/mjs/connectors/ComicK.mjs:
```diff
@@ -20,75 +21,87 @@ export default class ComicK extends Connector {
+                title += `Volume ${item.vol} `;
             }
-            title.push('Ch.', item.chap);
+            title += `Chapter ${item.chap}`;
```
Downloaded chapters are identified by their name, when a new naming pattern is applied, user who have downloaded chapters will no longer see them as downloaded...

starring94 replied:
fixed

27136773 commented on src/web/mjs/connectors/Copymanga.mjs:
```diff
@@ -0,0 +1,89 @@
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchJSON(request);
+        const pageCount = Math.ceil(data.results.total / 50);
+        for(let page = 1; page <= pageCount; page++) {
```
May use `page = 0; page < pageCount`

starring94 replied:
fixed

27136773 commented on src/web/mjs/connectors/Copymanga.mjs:
```diff
@@ -0,0 +1,89 @@
+    }
+
+    async _getMangasFromPage(page) {
+        const uri = new URL('/api/v3/comics?ordering=-datetime_updated&limit=50&offset=' + (page - 1) * 50, this.apiurl);
```
May use `page * 50` (see previous comment)

starring94 replied:
fixed

27136773 commented on src/web/mjs/connectors/Copymanga.mjs:
```diff
@@ -0,0 +1,89 @@
+        } else {
+            let mangaList = [];
+            const pageCount = Math.ceil(data.results.total / 500);
+            for(let page = 1; page <= pageCount; page++) {
```
May use `page = 0; page < pageCount`

starring94 replied:
fixed

27136773 commented on src/web/mjs/connectors/Copymanga.mjs:
```diff
@@ -0,0 +1,89 @@
+    }
+
+    async _getChaptersFromPage(manga, page) {
+        const uri = new URL(`/api/v3/comic/${manga.id}/group/default/chapters?limit=500&offset=${(page - 1) * 500}`, this.apiurl);
```
May use `page * 50` (see previous comment)

starring94 replied:
fixed

27136773 commented on src/web/mjs/connectors/Ichijin-Plus.mjs:
```diff
@@ -0,0 +1,13 @@
+        super.id = "ichijin-plus";
+        super.label = "一迅プラス (Ichijin Plus)";
+        this.tags = ["manga", "webtoon", "japanese"];
+        this.url = "https://ichijin-plus.com/";
```
May remove trailing slash `/`

starring94 replied:
fixed

27136773 commented on src/web/mjs/connectors/MangaItalia.mjs:
```diff
@@ -0,0 +1,69 @@
+        });
+    }
+
+    async _getPages(chapter) {
```
Use connector URI pattern to handle multiple HTML pages
See: https://github.com/manga-download/hakuneko/blob/master/src/web/mjs/connectors/MangaTown.mjs#L60-L76

starring94 replied:
This doesn't work, because it's not known how many pages there are in a chapter

27136773 commented on src/web/mjs/connectors/Mangabox.mjs:
```diff
@@ -0,0 +1,90 @@
+        for (let run = true; run;) {
+            const data = await this._getMangasFromPage(nextLink);
+            nextLink = data.nextLink;
+            nextLink ? mangaList.push(...data.chapters) : run = false;
```
Shouldn't the `mangas` list always be pushed, even if it has no next page?

starring94 replied:
Yes, fixed

27136773 commented on src/web/mjs/connectors/Mangabox.mjs:
```diff
@@ -0,0 +1,90 @@
+
+    async _getMangas() {
+        let mangaList = [];
+        let nextLink = undefined;
```
May initialize with `let nextLink = this.url + '/api/reader/episodes';`

starring94 replied:
changed

27136773 commented on src/web/mjs/connectors/ReadMNG.mjs:
```diff
@@ -17,73 +18,62 @@ export default class ReadMNG extends Connector {
+    }
+
+    async _getPages(chapter) {
+        const script = `
```
Looks like `WordPressMadara`, can't this template be used as base class?

starring94 replied:
It's actually WordPressMangastream

27136773 commented on src/web/mjs/connectors/ThreeHentai.mjs:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    async _getChapters(manga) {
+        return [ Object.assign({ language: '' }, manga) ];
```
May use spread operator if supported `return [ { ...manga, language: '' } ];`

starring94 replied:
fixed

27136773 commented on src/web/mjs/connectors/To-Corona-Ex.mjs:
```diff
@@ -19,47 +21,48 @@ export default class toCoronaEx extends Connector {
+        let request = new Request(uri, this.requestOptions);
+        let data = await this.fetchJSON(request);
+        return data.resources
+            .filter(chapter => chapter.episode_status == 'free_viewing')
```
Is a logged in user who purchased some chapters still able to see them?

starring94 replied:
Logging in manually doesn't change the chapter list, otherwise yes

27136773 commented on src/web/mjs/connectors/To-Corona-Ex.mjs:
```diff
@@ -19,47 +21,48 @@ export default class toCoronaEx extends Connector {
         return data;
     }
+
     prepareForDescramble(scrambledImageURL) {
```
May use async/await instead of promise chaining

starring94 replied:
changed

27136773 commented on src/web/mjs/connectors/templates/Tappytoon.mjs:
```diff
@@ -0,0 +1,72 @@
+        this.url = undefined;
+        this.baseURL = 'https://www.tappytoon.com';
+        this.apiurl = 'https://api-global.tappytoon.com';
+        this.lang = undefined;
```
This is used in some places, but never seems to be initialized correctly

starring94 replied:
It's defined correctly

27136773 commented on src/web/mjs/connectors/templates/Tappytoon.mjs:
```diff
@@ -0,0 +1,72 @@
+        let request = new Request(uri, this.requestOptions);
+        let data = await this.fetchDOM(request, 'script#__NEXT_DATA__');
+        let obj = JSON.parse(data[0].text).props.initialState.axios.headers;
+        this.requestOptions.headers.set('Authorization', obj['Authorization']);
```
Does these expire?
Will they still be the same when user login via manual website interaction?

starring94 replied:
The Authorization header contains a bearer token, and yes it will change.

27136773 reviewed (approved):
Generally looks good, some minor improvements possible

starring94 committed:
Updated connectors
* [3hentai] changed to spread operator
* [batoto] changed to app.items
* [comick] changed back naming scheme
* [copymanga] changed paging and reversed chapter list
* [ichijin-plus] removed trailing slash
* [mangabox] updated nextLink behaviour
* [readmng] changed to WPMangastream template
* [to-corona-ex] updated connector
-- changed to multi-page manga and chapter list
-- reversed chapter order
-- async descrambling

27136773 commented on src/web/mjs/connectors/To-Corona-Ex.mjs:
```diff
@@ -21,29 +21,61 @@ export default class ToCoronaEx extends Connector {
-            };
-        });
+        return {
+            chapters: data.resources.map(element => {
```
May be better to use `mangas:` instead of `chapters:`

starring94 replied:
fixed

27136773 reviewed (approved):
Looks good for merge now, except conflict in Bato that needs to be resolved

starring94 committed:
Merge branch 'master' into s-connectors3

starring94 committed:
Updated connector and test
* [test-e2e] [to-corona-ex] changed accessor from shift to pop
* [to-corona-ex] Changed from chapters to mangas on manga getter

starring94 committed:
Merge branch 's-connectors3' of https://github.com/starring94/hakuneko into s-connectors3
